Buildings/Applications/DHC/EnergyTransferStations/CoolingIndirect.mo

```diff
@@ -0,0 +1,287 @@
+    annotation(Dialog(group="Initialization",tab="Controller"));
+  parameter Real xi_start=0
+    "Initial or guess value value for integrator output (= integrator state)"
+    annotation (Dialog(group="Initialization",tab="Controller"));
+  parameter Real xd_start=0
+    "Initial or guess value for state of derivative block"
+    annotation (Dialog(group="Initialization",tab="Controller"));
+  parameter Real yCon_start=0
+    "Initial value of output from the controller"
+    annotation(Dialog(group="Initialization",tab="Controller"));
+  parameter Boolean reverseAction = true
+    "Set to true for throttling the water flow rate through a cooling coil controller"
+    annotation(Dialog(tab="Controller"));
+
+  Buildings.Fluid.HeatExchangers.PlateHeatExchangerEffectivenessNTU hex(
+    redeclare package Medium1 = Medium,
+    redeclare package Medium2 = Medium,
+    m1_flow_nominal=m1_flow_nominal,
+    m2_flow_nominal=m2_flow_nominal,
+    dp1_nominal=dp1_nominal,
+    dp2_nominal=dp2_nominal,
+    configuration=Buildings.Fluid.Types.HeatExchangerConfiguration.CounterFlow,
+    use_Q_flow_nominal=true,
+    Q_flow_nominal=Q_flow_nominal,
+    T_a1_nominal=T_a1_nominal,
+    T_a2_nominal=T_a2_nominal) "Indirect cooling heat exchanger"
+    annotation (Placement(transformation(
+        extent={{10,10},{-10,-10}},
+        rotation=180,
+        origin={30,0})));
+
+  Modelica.Blocks.Interfaces.RealInput TSet "Setpoint temperature"
+    annotation (Placement(transformation(extent={{-140,-20},{-100,20}})));
+
+  Buildings.Controls.Continuous.LimPID con(
+    controllerType=Modelica.Blocks.Types.SimpleController.PID,
+    final k=k,
+    Td=Td,
+    final yMax=yMax,
+    final yMin=yMin,
+    final Ti=Ti,
+    wp=wp,
+    wd=wd,
+    Ni=Ni,
+    Nd=Nd,
+    final initType=Modelica.Blocks.Types.InitPID.InitialOutput,
+    xi_start=xi_start,
+    xd_start=xd_start,
+    final y_start=0,
+    final reverseAction=reverseAction) "Controller"
+    annotation (Placement(transformation(extent={{-90,-10},{-70,10}})));
+
+  Buildings.Fluid.Sensors.TemperatureTwoPort senTDisSup(
+    redeclare package Medium = Medium,
+    m_flow_nominal=m1_flow_nominal)
+    "District-side (primary) supply temperature sensor"
+    annotation (Placement(transformation(
+        extent={{-10,-10},{10,10}},
+        rotation=0,
+        origin={-80,60})));
+
+  Buildings.Fluid.Sensors.TemperatureTwoPort senTDisRet(
+    redeclare package Medium = Medium,
+    m_flow_nominal=m1_flow_nominal)
+    "District-side (primary) return temperature sensor"
+    annotation (Placement(transformation(
+        extent={{-10,-10},{10,10}},
+        rotation=0,
+        origin={80,60})));
+
+  Modelica.Blocks.Sources.RealExpression powCal(
+    y=senMasFlo.m_flow*cp*(senTDisRet.T - senTDisSup.T))
+    "Calculated power demand"
+    annotation (Placement(transformation(extent={{-100,-100},{-20,-80}})));
+
+  Modelica.Blocks.Interfaces.RealOutput Q_flow(
```

Comment:
make sure the Modelica script text in following order:

1. `extends`
2. `parameter` section
3. `connector` section
4. classes instantiation
5. `protected` section
6. equation section
So, move the `Interfaces.xxx` up, to follow the parameter section

Reply:
Just to confirm before I make changes, it appears most models in the Modelica Standard and Buildings libraries follow:
1. `extends`
2. `parameter` section
3. classes instantiation
4. `protected` section
5. **`connector` section**
6. equation section

@JayHuLBL Could you please confirm the correct order?

Reply:
I will move the two Interfaces.xxx (`Q_flow` and `Q`) to be at the top of the "classes instantiation" (blocks) section, along with the (`TSet` Interfaces). My main question is if I need to also move the `connector` section for all models as well, as indicated in your order list. Thanks

Reply:
The `connector section` in my comment means the `Interfaces.xxx`. So
1. `extends`
2. `parameter` section
3. `connector` section, like `.Interfaces.xxx`.
4. classes instantiation
5. `protected` section
6. equation section

The `connect(xx,xx)` should be in the equation section as it is now.

Reply:
I see, thank you for clarifying.